Tests/DataXScenarios/DataXScenarios/InteractiveQueryAndSchemaGenScenarios.cs

```diff
@@ -0,0 +1,125 @@
+﻿// *********************************************************************
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License
+// *********************************************************************
+using ScenarioTester;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DataX.ServerScenarios
+{
+    /// <summary>
+    /// Partial class defined such that the steps can be defined for the job 
+    /// </summary>
+    public partial class DataXHost
+    {
+        [Step("inferSchema")]
+        public static StepResult InferSchema(ScenarioContext context)
+        {
+            var baseAddress = $"{context[Context.ServiceUrl] as string}/api/DataX.Flow/Flow.SchemaInferenceService/inputdata/inferschema";
+            context[Context.InferSchemaInputJson] = $"{{\"name\": \"{context[Context.FlowName] as string}\", \"userName\": \"{context[Context.FlowName] as string}\", \"eventhubConnectionString\": \"{context[Context.EventhubConnectionString] as string}\", \"eventHubNames\": \"{context[Context.EventHubName] as string}\", \"inputType\": \"iothub\", \"seconds\": \"{context[Context.Seconds] as string}\", \"databricksToken\": \"{context[Context.DataBricksToken] as string}\"}}";
+            string jsonResult = Request.Post(baseAddress,
+                    RequestContent.EncodeAsJson(
```

Comment:
For these StepReult implementations, the following pattern is repeated and looks like this can be refactored.
Generate baseAddress
Compose body for Post
Call Post/Get
Get the result

Reply:
Not  refactoring this at this point for readability and not creating a wrapper on top of ScenarioTester.